packages/amazonq/src/app/inline/completion.ts

```diff
@@ -203,52 +204,64 @@ export class AmazonQInlineCompletionItemProvider implements InlineCompletionItem
 
-            // make service requests if it's a new session
-            await this.recommendationService.getAllRecommendations(
-                this.languageClient,
-                document,
-                position,
-                context,
-                token
-            )
-        }
-        // get active item from session for displaying
-        const items = this.sessionManager.getActiveRecommendation()
-        const session = this.sessionManager.getActiveSession()
-        if (!session || !items.length) {
-            return []
-        }
-        const editor = window.activeTextEditor
-        for (const item of items) {
-            item.command = {
-                command: 'aws.amazonq.acceptInline',
-                title: 'On acceptance',
-                arguments: [
-                    session.sessionId,
-                    item,
-                    editor,
-                    session.requestStartTime,
+                // make service requests if it's a new session
+                await this.recommendationService.getAllRecommendations(
+                    this.languageClient,
+                    document,
+                    position,
+                    context,
+                    token
+                )
+            }
+            // get active item from session for displaying
+            const items = this.sessionManager.getActiveRecommendation()
+            const session = this.sessionManager.getActiveSession()
+            if (!session || !items.length) {
+                return []
+            }
+            const editor = window.activeTextEditor
+            for (const item of items) {
+                item.command = {
+                    command: 'aws.amazonq.acceptInline',
+                    title: 'On acceptance',
+                    arguments: [
+                        session.sessionId,
+                        item,
+                        editor,
+                        session.requestStartTime,
+                        position.line,
+                        session.firstCompletionDisplayLatency,
+                    ],
+                }
+                ReferenceInlineProvider.instance.setInlineReference(
                     position.line,
-                    session.firstCompletionDisplayLatency,
-                ],
+                    item.insertText as string,
+                    item.references
+                )
+                ImportAdderProvider.instance.onShowRecommendation(document, position.line, item)
             }
-            ReferenceInlineProvider.instance.setInlineReference(
-                position.line,
-                item.insertText as string,
-                item.references
-            )
-            ImportAdderProvider.instance.onShowRecommendation(document, position.line, item)
+            // exposed only for testing
+            vsCodeState.isRecommendationsActive = false
```

Comment:
We should be able to delete this line since it is in the finally

Reply:
you need to enable it before inline completions responds, I have no idea why though